[Yan Hwee] iP

yanhwee opened the pull request:
### Chatbot Gray 👋

```java
public class Gray {
   public static void main(String[] args) {
      System.out.println("Hello! I am Chatbot Gray!);
   }  
}
```

> Hello! I am Chatbot Gray!

_Features:_
1. Add tasks
2. Delete tasks
3. List tasks
4. Save tasks
5. Load tasks

Types of tasks supported:
- `Todo`
- `Deadline`
- `Event`

**Future features:**
- [x] GUI
- [ ] Natural language processing like [ChatGPT](https://openai.com/index/chatgpt/)

~~Some strikethrough text~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yanhwee committed:
Rename chatbot to gray

yanhwee committed:
Add echo feature

yanhwee committed:
Add list feature

yanhwee committed:
Add mark/unmark feature

yanhwee committed:
Add new tasks (todos, events, deadlines) feature

yanhwee committed:
Add text ui testing

yanhwee committed:
Add handle errors

yanhwee committed:
Add delete feature

yanhwee committed:
Add save feature

yanhwee committed:
Merge branch 'branch-Level7'

yanhwee committed:
Add parse datetime feature

yanhwee committed:
Merge branch 'branch-Level8'

yanhwee committed:
Refactor more OOP

yanhwee committed:
Merge branch 'branch-A-MoreOOP'

yanhwee committed:
Refactor add packages

yanhwee committed:
Merge branch 'branch-A-Packages'

yanhwee committed:
Merge remote-tracking branch 'remotes/origin/add-gradle-support'

yanhwee committed:
Change package of TaskList from gray.task to gray

yanhwee committed:
Add junit test cases

yanhwee committed:
Merge branch 'branch-A-JUnit'

yanhwee committed:
Configure to package app as JAR file

yanhwee committed:
Merge branch 'branch-A-Jar'

yanhwee committed:
Add javadoc comments

yanhwee committed:
Tweak code to comply with coding standard

yanhwee committed:
Add find feature

yanhwee committed:
Merge branch 'branch-A-JavaDoc'

yanhwee committed:
Merge branch 'branch-A-CodingStandard'

yanhwee committed:
Merge branch 'branch-Level9'

yanhwee committed:
Delete data files

yanhwee committed:
Update .gitignore to ignore data files

yanhwee committed:
Merge branch 'fix-gitignore'

yanhwee committed:
Update README

yanhwee committed:
Add checkstyle

yanhwee committed:
Fix style check errors and warnings

yanhwee committed:
Merge branch 'branch-A-CheckStyle'

yanhwee committed:
Add GUI

yanhwee committed:
Merge branch 'branch-Level-10'

Zeepheru commented on src/main/java/gray/gui/MainWindow.java:
```diff
@@ -0,0 +1,85 @@
+package gray.gui;
+
+import static gray.Constants.FILEPATH_VIEW_MAIN_WINDOW;
+
+import java.io.IOException;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import gray.Main;
+import gray.gui.DialogBox;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
```
Consistent ordering of import statements.

Zeepheru commented on src/main/java/gray/command/FindCommand.java:
```diff
@@ -0,0 +1,38 @@
+        if (matchingTasks.isEmpty()) {
+            return "No matches found";
+        }
+        String matches = String.join("\n",
+                IntStream.range(0, matchingTasks.size())
+                        .mapToObj(i -> String.format(
+                                "%d. %s",
+                                i + 1,
+                                matchingTasks.get(i)))
+                        .collect(Collectors.toList()));
+        return String.format(
+                "Here are the matching tasks in your list:\n%s",
+                matches);
```
Correct indentation for wrapped lines.

Zeepheru commented on src/main/java/gray/Tasks.java:
```diff
@@ -0,0 +1,91 @@
+        }
+    }
+
+    public int size() {
+        return tasks.size();
+    }
```
Very very minor nitpick. I understand the use of `size()` as a natural continuation of `ArrayList::size`, since the tasks are stored in the class as an `ArrayList` anyway. But maybe naming the method differently, e.g. `tasks.count()` would be clearer than `tasks.size()`?

Zeepheru commented on src/main/java/gray/Constants.java:
```diff
@@ -0,0 +1,14 @@
+ */
+public class Constants {
+    // Resource files
+    public static final String FILEPATH_VIEW_MAIN_WINDOW = "/view/MainWindow.fxml";
+    public static final String FILEPATH_IMAGE_GRAY = "/images/gray.png";
+    public static final String FILEPATH_IMAGE_USER = "/images/user.png";
+
+    // Non-resource files
+    public static final String FILEPATH_DATA_TASKS = "./data/saveTasks";
```
Good use of SCREAMING_SNAKE_CASE for constants.

Zeepheru commented on src/main/java/gray/gui/DialogBox.java:
```diff
@@ -0,0 +1,39 @@
+    private Label label;
+
+    /**
+     * Constructs a dialog box for the GUI.
+     * @param image
```
Should leave a line between the description and parameter section.

Zeepheru reviewed (commented):
LGTM! Overall good code styling, naming that follows the conventions in the style guide.
Just minor nits that I only wrote to meet the comment count quota 😅

yu-sutong reviewed (commented):
All method and class naming is correct. Well done!

yanhwee committed:
Add assertions to check that filepaths are valid

There is no way to automatically check if filepaths
(especially resource files) provided in Constants.java
are valid or existent.

Therefore, validating these files need to be done
manually, otherwise the validity of these filepaths
are left unknown until runtime.

Thus, provide a function that automates these checks
(implemented in the main function of Constants.java).

As this is the only validation done by Constants.java,
it is adequate and convenient to execute the check
through the main function of Constants.java.

yanhwee committed:
Refactor method names

Some method names are in British english.

I think that they should be in American english, for standardization
purposes, as most (if not all) java utils libraries uses American
spellings.

Thus, refactor and rename these method names that has British spellings
into their correspounding American spellings.

It is done this way for standardization purposes.

yanhwee committed:
Refactor loops to stream for list command

The current implementation of the list command to generate the display
string of the task list uses a for loop.

The for loop implementation requires the explicit use of a StringJoiner
and requires explicit management of how the strings are joined.

Refactor the loop to use java streams instead.

The resultant code that uses the stream is simplier (in my opinion),
without the explicit use of StringJoiner and explict adding of strings
to the StringJoiner object. Rather, an alternate approach to declare
that these stream of strings are to be joined together via the
`Stream.collect` and `Collectors.joining` methods.

yanhwee committed:
Merge pull request #1 from yanhwee/branch-A-Assertions

Add assertions to check that filepaths are valid

yanhwee committed:
Merge branch 'master' into branch-A-CodeQuality

yanhwee committed:
Merge branch 'master' into branch-A-Streams

yanhwee committed:
Merge pull request #2 from yanhwee/branch-A-CodeQuality

Refactor method names

yanhwee committed:
Merge branch 'master' into branch-A-Streams

yanhwee committed:
Merge pull request #3 from yanhwee/branch-A-Streams

Refactor loops to stream for list command

yanhwee committed:
Add help feature

yanhwee committed:
Merge branch 'branch-C-Help'

yanhwee committed:
Improve GUI

yanhwee committed:
Merge branch 'branch-A-BetterGui'

yanhwee committed:
Fix bug not saving when marking/unmarking task

yanhwee committed:
Add product name to GUI window

yanhwee committed:
Fix minor typos

yanhwee committed:
Update docs README

yanhwee committed:
Merge branch 'branch-A-UserGuide'

yanhwee committed:
Update message for listing of tasks

yanhwee committed:
Update build.gradle

yanhwee committed:
Fix mistake in user guide
